10up/ElasticPress
# New Filter ep_highlight_number_of_fragments.

Add to ./includes/classes/Feature/Search/Search.php a new filter to set the maximum number of fragments to highlight.



### Description of the Change

Added a filter that allows the dev to set the max number of fragments to highlight. This is just a convenience filter that relieves us from having to modify the formatted args directly. This maintains the current default setting of zero fragments.


Closes #3579 

### How to test the Change

Add this to a function:

```
// set number of fragments to 10 for post_content and post_excerpt fields (zero is the default)
add_filter( 'ep_highlight_number_of_fragments', function( $number_of_fragments, $field ) {
    if ( in_array( $field, [ 'post_content', 'post_excerpt' ] ) ) {
        $number_of_fragments = 10;
    }
    return $number_of_fragments;
} );

```
Check the query body "highlight" section for the "number_of_fragments" setting.

### Changelog Entry

> Added - New feature
> Changed - Existing functionality
> Deprecated - Soon-to-be removed feature
> Removed - Feature
> Fixed - Bug fix
> Security - Vulnerability


### Credits

Props @username, @username2, ...
Props @dgnorrod

### Checklist:


- [x ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
